Library/Homebrew/compat/formula_support.rb

```diff
@@ -0,0 +1,18 @@
+require "formula_support"
+
+class KegOnlyReason
+  alias _to_s to_s
+
+  def to_s
+    case @reason
+    when :provided_by_osx
+      odeprecated "keg_only :provided_by_osx", "keg_only :provided_by_macos"
+      @reason = :provided_by_macos
+    when :shadowed_by_osx
+      odeprecated "keg_only :shadowed_by_osx", "keg_only :shadowed_by_macos"
+      @reason = :shadowed_by_macos
+    end
+
+    _to_s
+  end
```

Comment:
`Module#prepend` would probably be nicer than an aliasing hack.

Reply:
@alyssais Yeh, good call. I'm using this for now for consistency with the rest of `compat` but I'd ❤️ a refactoring PR after this to use `Module#prepend` everywhere instead.